pytest_plugins/fixture_markers.py

```diff
@@ -6,10 +6,14 @@
 
 def pytest_generate_tests(metafunc):
     if 'rhel_contenthost' in metafunc.fixturenames:
-        rhel_parameters = [
-            dict(workflow=settings.content_host.deploy_workflow, rhel_version=ver)
-            for ver in settings.content_host.rhel_versions
-        ]
+        rhel_parameters = []
+        for ver in settings.content_host.rhel_versions.keys():
+            # prefer nick-specific deploy workflow before using the default one
+            workflow = (
+                settings.content_host.rhel_versions[ver].get('deploy_workflow')
+                or settings.content_host.deploy_workflow
```

Comment:
non-blocking and black will likely adjust formatting
```suggestion
                settings.content_host.rhel_versions[ver].get('deploy_workflow', settings.content_host.deploy_workflow)
```